junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/CsvSource.java

```diff
@@ -105,4 +105,16 @@
 	@API(status = EXPERIMENTAL, since = "5.6")
 	String[] nullValues() default {};
 
+	/**
+	 * The maximum characters of per CSV column allowed.
+	 *
+	 * <p>To enable auto-expansion of the internal array, set this property to -1
+	 *
+	 * <p>Defaults to {@code 4096}.
+	 *
+	 * @since 5.7
+	 */
+	@API(status = EXPERIMENTAL, since = "5.7")
+	int maxCharsPerColumn() default 4096;
```

Comment:
**Team Decision**: Apply same changes as for `@CsvFileSource`.

Reply:
Updated